SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -642,12 +665,51 @@ void BoxSorting::program(SingleSwitchProgramEnvironment& env, SwitchControllerCo
                         }
                         boxes_data[get_index(box_nb, row, column)]->ot_id = ot_id;
 
-                        // NOTE edit when adding new struct members (detections go here likely)
-
                         // level_box
-                        // ot_box
+                        int level = OCR::read_number_waterfill(env.console, extract_box_reference(screen, level_box), 0xff000000, 0xff7f7f7f);
+                        if (level < 0 || level > 100) {

```

Comment:
<= 0 ?
And have you tried with eggs ? (I forgot what the program does when there is eggsà